infra/notifications/src/emailSend.js

```diff
@@ -19,10 +19,133 @@ if (!process.env.SENDGRID_FROM_EMAIL) {
   logger.warn('Warning: SENDGRID_FROM_EMAIL env var is not set')
 }
 
+const {
+  getMessageFingerprint,
+  isNotificationDupe,
+  logNotificationSent
+} = require('./dupeTools')
+
 //
-// Email notifications
+// Email notifications for Messages
 //
-async function emailSend(
+async function messageEmailSend(receivers, sender, config) {
+  if (!receivers) throw 'receivers not defined'
+  if (!sender) throw 'sender not defined'
+
+  // Load email template
+  const templateDir = `${__dirname}/../templates`
+
+  // Standard template for HTML emails
+  const emailTemplateHtml = _.template(
+    fs.readFileSync(`${templateDir}/emailTemplate.html`).toString()
+  )
+  // Standard template for text emails
+  const emailTemplateTxt = _.template(
+    fs.readFileSync(`${templateDir}/emailTemplate.txt`).toString()
+  )
+
+  const messageSender = Identity.findOne({
+    where: {
+      ethAddress: sender
+    }
+  })
+  const messageReceivers = Identity.findAll({
+    where: {
+      ethAddress: {
+        [Op.or]: receivers
+      }
+    }
+  })
+  Promise.all([messageSender, messageReceivers]).then(
+    ([senderIdentity, receiversIdentities]) => {
+      console.log(senderIdentity)
+      receiversIdentities.forEach(async s => {
```

Comment:
This won't work for cases where the person has never published an identity as messageReceivers will be empty. The same thing applies to the mobile code. Not sure how we want to handle that. I know we want to encourage people to have an identity but I still think notifications from unnamed users should work.

Reply:
Excellent point. Added code to follow the convention of using the eth address where a name is not available. 
Commit: 32d4a7e5a736cbfcab69cd94d60ee03e53538a93

Reply:
Users without any identity published can still be contacted - via messaging. All mobile users will have messaging enabled and be able to receive push notifications but have no identity published when the app first opens. I think it'll be reasonably common for someone to open the mobile app, see something they want to buy, and message the user without publishing an identity.

Reply:
Gotcha, true. We're okay for email (as in this particular case of `emailSend.js`) but for mobilepush with auto-enable messaging this will be an issue. I'll figure out the sequeleze for getting back a NULL in case of no identity.

Reply:
Ok, I pulled out the not-yet-used-anyway stuff around sender and receiver identities. This plain-jane approach should work for all mobile users: 2149e346aa60c477195b0c427fe5532c9c7ed478